RadarSDK/RadarAPIClient.m

```diff
@@ -244,106 +333,66 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
-                            }
-                            [RadarState setPlaceId:placeId];
-                            
-                            NSMutableArray *regionIds = [NSMutableArray new];
-                            if (user.country) {
-                                [regionIds addObject:user.country._id];
-                            }
-                            if (user.state) {
-                                [regionIds addObject:user.state._id];
-                            }
-                            if (user.dma) {
-                                [regionIds addObject:user.dma._id];
-                            }
-                            if (user.postalCode) {
-                                [regionIds addObject:user.postalCode._id];
-                            }
-                            [RadarState setRegionIds:regionIds];
-                            
-                            NSMutableArray *beaconIds = [NSMutableArray new];
-                            if (user.beacons) {
-                                for (RadarBeacon *beacon in user.beacons) {
-                                    [beaconIds addObject:beacon._id];
-                                }
-                                
-                            }
-                            [RadarState setBeaconIds:beaconIds];
-                        }
+        NSMutableArray *trackParams = [RadarReplay arrayForReplays:replays];
+        [trackParams addObject:params];
         
-                        if (events && user) {
-                            [RadarSettings setId:user._id];
-                            
-                            // if user was on a trip that ended server side, restore previous tracking options
-                            if (!user.trip && [RadarSettings tripOptions]) {
-                                [[RadarLocationManager sharedInstance] restartPreviousTrackingOptions];
-                                [RadarSettings setTripOptions:nil];
-                            }
-
-                            if (location) {
-                                [[RadarDelegateHolder sharedInstance] didUpdateLocation:location user:user];
-                            }
-
-                            if (events.count) {
-                                [[RadarDelegateHolder sharedInstance] didReceiveEvents:events user:user];
-                            }
+        NSString *bulkUrl = [NSString stringWithFormat:@"%@/v1/track/bulk", host];
+        bulkUrl = [bulkUrl stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
 
-                            return completionHandler(RadarStatusSuccess, res, events, user, nearbyGeofences, meta);
-                        }
-
-                        [[RadarDelegateHolder sharedInstance] didFailWithStatus:status];
+        NSDictionary *trackParamsDict = @{@"tracks": trackParams};
+        
+        [self.apiHelper requestWithMethod:@"POST"
```

Comment:
I think it will be a little simpler if we can just have a single `[self.apiHelper requestWithMethod ...]` call and above it pick `/track` vs `/track/bulk` and set up the other params.

Then when we're dealing with failed requests below we can check `options.replay` and handle  `RadarTrackingOptionsReplayStops` vs `RadarTrackingOptionsReplayAll`.